### DISCUSSION TITLE
What's the difference?

### DISCUSSION THREAD

**Discussion by yuriy23** [espressif/arduino-esp32#10289] [General]
What is the difference in terms of speed, occupied RAM, and so on?

  SPIClass * hspi = NULL;
void TaskSPI(void *pvParameters)
{
  hspi = new SPIClass(HSPI);
  hspi->begin(36, 35, 21, 47); //SCLK, MISO, MOSI, SS  
}

    ** **Or so****

  #define VSPI 3
  SPIClass vspi = SPIClass(VSPI);
void TaskSPI(void *pvParameters)
{
SPIClass vspi = SPIClass(VSPI);
vspi.begin(36, 35, 21, 47); //SCLK, MISO, MOSI, SS
}

**Comment by me-no-dev** [comment]
None. You are just using another SPI port

**Reply by yuriy23** [reply]
the question is not about the port number, but about a different entry..
vspi->begin
vspi.begin

**Reply by me-no-dev** [reply]
`.` you access the object directly, `->` you access it by reference to it's location in memory. No difference in speed or anything else.

**Reply by yuriy23** [reply]
thanks
